e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/add-to-an-existing-payment.spec.js

```diff
@@ -149,4 +153,80 @@ context(`${PDC_TEAMS.PDC_RECONCILE} users can add fee records to existing paymen
+      .withPaymentCurrency(PAYMENT_CURRENCY)
+      .withFeesPaidToUkefForThePeriod(300)
+      .withFeesPaidToUkefForThePeriodCurrency('EUR')
+      .withPaymentExchangeRate(0.5)
+      .withStatus('TO_DO')
+      .build();
+    const fourthFeeRecord = FeeRecordEntityMockBuilder.forReport(report)
+      .withId(FEE_RECORD_ID_FOUR)
+      .withFacilityId('44444444')
+      .withExporter('Exporter 4')
+      .withPaymentCurrency(PAYMENT_CURRENCY)
+      .withFeesPaidToUkefForThePeriod(400)
+      .withFeesPaidToUkefForThePeriodCurrency('EUR')
+      .withPaymentExchangeRate(0.5)
+      .withPayments([firstPayment])
+      .withStatus('DOES_NOT_MATCH')
+      .build();
+
+    cy.task(NODE_TASKS.REMOVE_ALL_UTILISATION_REPORTS_FROM_DB);
+    cy.task(NODE_TASKS.REMOVE_ALL_PAYMENTS_FROM_DB);
+    cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [report]);
+    cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [firstFeeRecord, secondFeeRecord, thirdFeeRecord, fourthFeeRecord]);
+
+    pages.landingPage.visit();
+    cy.login(USERS.PDC_RECONCILE);
+
+    cy.visit(`utilisation-reports/${REPORT_ID}`);
+    getFeeRecordCheckbox([FEE_RECORD_ID_ONE], PAYMENT_CURRENCY, FEE_RECORD_STATUS.TO_DO).check();
+    getFeeRecordCheckbox([FEE_RECORD_ID_THREE], PAYMENT_CURRENCY, FEE_RECORD_STATUS.TO_DO).check();
+
+    cy.get('[type="submit"]').contains('Add a payment').click();
+    cy.url().should('eq', relative(`/utilisation-reports/${REPORT_ID}/add-payment`));
+    cy.get('[type="submit"]').contains('Add reported fees to an existing payment').click();
+    cy.url().should('eq', relative(`/utilisation-reports/${REPORT_ID}/add-to-an-existing-payment`));
+
+    pages.utilisationReportAddToAnExistingPaymentPage.backLink().click();
+    getFeeRecordCheckbox([FEE_RECORD_ID_ONE], PAYMENT_CURRENCY, FEE_RECORD_STATUS.TO_DO).should('be.checked');
+    getFeeRecordCheckbox([FEE_RECORD_ID_THREE], PAYMENT_CURRENCY, FEE_RECORD_STATUS.TO_DO).should('be.checked');
+    getFeeRecordCheckbox([FEE_RECORD_ID_TWO, FEE_RECORD_ID_FOUR], PAYMENT_CURRENCY, FEE_RECORD_STATUS.DOES_NOT_MATCH).should('not.be.checked');
+
+    cy.get('[type="submit"]').contains('Add a payment').click();
+    cy.url().should('eq', relative(`/utilisation-reports/${REPORT_ID}/add-payment`));
+    cy.get('[type="submit"]').contains('Add reported fees to an existing payment').click();
+    cy.url().should('eq', relative(`/utilisation-reports/${REPORT_ID}/add-to-an-existing-payment`));
+
+    pages.utilisationReportAddToAnExistingPaymentPage.cancelLink().click();
+    getFeeRecordCheckbox([FEE_RECORD_ID_ONE], PAYMENT_CURRENCY, FEE_RECORD_STATUS.TO_DO).should('be.checked');
+    getFeeRecordCheckbox([FEE_RECORD_ID_THREE], PAYMENT_CURRENCY, FEE_RECORD_STATUS.TO_DO).should('be.checked');
+    getFeeRecordCheckbox([FEE_RECORD_ID_TWO, FEE_RECORD_ID_FOUR], PAYMENT_CURRENCY, FEE_RECORD_STATUS.DOES_NOT_MATCH).should('not.be.checked');
+  });
+
+  function getFeeRecordCheckbox(feeRecordIds, paymentCurrency, status) {
```

Comment:
Looks like there are very similar `getFeeRecordCheckbox` functions used in other places - could this be a DRY cypress command?